src/pathToVictory/services/viability.service.ts

```diff
@@ -116,52 +130,291 @@ export class ViabilityService {
-        score += 0.5
+  // This function is a wrapper to call the ported python code (getInitialViabilityScore)
+  // It is separated so that we can easily update the getInitialViabilityScore function.
+  calculateNewViabilityScore(
+    state: string,
+    officeLevel: string,
+    officeType: string,
+    electionPartisanship: boolean,
+    seats: number,
+    incumbent: boolean,
+    opponents: number,
+    openSeat: boolean,
+  ): ViabilityScore {
+    try {
+      this.logger.log(
+        `Calculating new viability score for ${state}, ${officeLevel}, ${officeType}`,
+      )
+
+      const { probOfWin, rating } = this.calculateInitialViability(
+        state,
+        officeLevel,
+        officeType,
+        electionPartisanship,
+        seats,
+        incumbent,
+        opponents,
+        openSeat,
+      )
+
+      // Format the viability score according to the existing interface
+      const viabilityScore: ViabilityScore = {
+        level: officeLevel,
+        isPartisan: electionPartisanship,
+        isIncumbent: incumbent,
+        isUncontested: opponents === 0,
+        candidates: opponents + 1, // Total candidates including the campaign
+        seats: seats,
+        candidatesPerSeat: (opponents + 1) / seats,
+        score: rating,
+        probOfWin,
       }
+
+      return viabilityScore
+    } catch (error: unknown) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown error'
+      const errorStack = error instanceof Error ? error.stack : undefined
+      this.logger.error(
+        `Error calculating new viability score: ${errorMessage}`,
+        errorStack,
+      )
+      throw error
+    }
+  }
+
+  /**
+   * This function has been ported from python and will need to be updated
+   * Calculates the initial viability score for a campaign based on various factors
+   * @param state - The state where the election is taking place
+   * @param officeLevel - The level of office (federal, regional, state, county, city, town, local)
+   * @param officeType - The type of office being sought
+   * @param electionPartisanship - Whether the election is partisan
+   * @param seats - Number of seats available
+   * @param incumbent - Whether the candidate is an incumbent
+   * @param opponents - Number of opponents
+   * @param openSeat - Open seat is when the number of incumbents running is less than the number of seats available
+   * @returns An object containing the probability exponent, probability of winning, and rating (1-5)
+   */
+  calculateInitialViability(
+    state: string,
+    officeLevel: string,
+    officeType: string,
+    electionPartisanship: boolean,
+    seats: number,
+    incumbent: boolean,
+    opponents: number,
+    openSeat: boolean,
+  ): { probExponent: number; probOfWin: number; rating: number } {
+    // Implements a logistic regression term by term with co-efficients hard-coded
+    let probExponent = -1.70059412
```

Comment:
it will be nice to have all of these random numbers in an object or enum so we can be more descriptive and understand what they mean.